core/src/main/java/org/elasticsearch/monitor/os/OsProbe.java

```diff
@@ -180,6 +188,294 @@ public short getSystemCpuPercent() {
         return Probes.getLoadAndScaleToPercent(getSystemCpuLoad, osMxBean);
     }
 
+    /**
+     * Reads all lines of a file and logs any {@link IOException}.
+     *
+     * @param path path to the file to read
+     * @return all the line or an empty list if an {@link IOException}
+     * occurred
+     */
+    private List<String> readAllLines(final Path path) {
+        try {
+            return Files.readAllLines(path);
+        } catch (final IOException e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("error reading " + path, e);
+            }
+            return Collections.emptyList();
+        }
+    }
+
+    /**
+     * Reads a file containing a single line and logs any
+     * {@link IOException}.
+     *
+     * @param path path to the file to read
+     * @return the single line or {@code null} if an
+     * {@link IOException} occurred
+     */
+    private String readSingleLine(final Path path) {
+        try {
+            final List<String> lines = Files.readAllLines(path);
+            assert lines != null && lines.size() == 1;
+            return lines.get(0);
+        } catch (final IOException e) {
```

Comment:
do we really need this leniency here everywhere? I mean can't we catch the expection on a higher level and skip the entire stats? I feel this kind of stuff will just hide problems from us and adds lots of unnecessary lenient code?